src/jsdoc_transformer.ts

```diff
@@ -130,29 +130,30 @@ export function maybeAddHeritageClauses(
       return null;
     }
 
-    // Resolve any aliases to the underlying type.
-    if (sym.flags & ts.SymbolFlags.TypeAlias) {
-      // It's implementing a type alias.  Follow the type alias back
-      // to the original symbol to check whether it's a type or a value.
-      const type = mtt.typeChecker.getDeclaredTypeOfSymbol(sym);
-      if (!type.symbol) {
-        // It's not clear when this can happen.
-        mtt.debugWarn(decl, `could not get type of symbol: ${expr.getText()}`);
-        return null;
-      }
-      sym = type.symbol;
-    }
-    if (sym.flags & ts.SymbolFlags.Alias) {
-      sym = mtt.typeChecker.getAliasedSymbol(sym);
+    // localSym can be an alias (e.g. an imported symbol); dealias it before
+    // examining flags, like whether it's a type or a value.  But keep the
+    // original symbol around, because we want to only reference the local
+    // name in any emitted reference.
+    let trueSym = localSym;
+    if (trueSym.flags & ts.SymbolFlags.Alias) {
+      trueSym = mtt.typeChecker.getAliasedSymbol(trueSym);
     }
 
     const typeTranslator = mtt.newTypeTranslator(expr.expression);
-    if (typeTranslator.isBlackListed(sym)) {
+    if (typeTranslator.isBlackListed(trueSym)) {
       // Don't emit references to blacklisted types.
       return null;
     }
 
-    if (sym.flags & ts.SymbolFlags.Class) {
+    if (trueSym.flags & ts.SymbolFlags.TypeAlias) {
+      // It's implementing a type alias.  This is illegal in Closure.
+      // We cannot just dereference the type alias, because it might refer
+      // to a symbol that hasn't been imported.
+      mtt.debugWarn(decl, `omitting heritage of type alias: ${expr.getText()}`);
```

Comment:
in another change, we'll need to adapt this code to collect all the heritage clauses we were unable to emit, and then emit the "incorrect assignment" pattern for all of them, right?